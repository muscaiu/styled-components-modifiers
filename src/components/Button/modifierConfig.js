
export default {
  primary: ({ theme }) => `
      background-color: ${theme.colors.primary};
    `,
  secondary: ({ theme }) => `
      background-color: ${theme.colors.secondary};
    `,
  disabled: () => `
      opacity: 50%;
      pointer-events: none;
    `,
};
