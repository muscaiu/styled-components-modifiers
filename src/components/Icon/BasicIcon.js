import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import FontAwesome from 'react-fontawesome';
// import faStyles from 'font-awesome/css/font-awesome.css'
import { omit } from 'lodash';

function UnstyledIcon(props) {
    const iconProps = omit(props, ['modifiers']);
    console.log('iconProps:', iconProps)
    //return  <FontAwesome {...iconProps} />
    return (
        <FontAwesome
            className="super-crazy-colors"
            name="home"
            // cssModule={faStyles}
            // size="2x"
            // spin
            style={{ textShadow: '0 1px 0 rgba(0, 0, 0, 0.1)' }}
            {...iconProps}
        />
    )
}

const BasicIcon = styled(UnstyledIcon)`
  font-size: 1rem;
`;

BasicIcon.propTypes = {
    name: PropTypes.string.isRequired
};

export default BasicIcon;
