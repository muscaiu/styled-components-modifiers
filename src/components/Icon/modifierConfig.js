export default {
    left: () => `
    padding-right: 0.5rem;
  `,
    right: () => `
    padding-left: 0.5rem;
  `,
};