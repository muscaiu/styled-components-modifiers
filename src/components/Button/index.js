import React from 'react';
import styled from 'styled-components';
import { rgba, lighten } from 'polished';
import { applyStyleModifiers } from 'styled-components-modifiers';

import withTheme from 'hoc/withTheme';
import modifierConfig from './modifierConfig'
// import Icon from '../Icon';
// import Text from '../Text';

const StyledButton = styled.button`
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFF;
  cursor: pointer;
  font-size: 1rem;
  padding: 0.5rem 1.5rem;
  outline: none;
  transition: all 200ms ease;

  &:hover,
  &:focus {
    background-color: ${lighten(0.05, '#7D7D7D')};
    box-shadow: 0 8px 8px 0 ${rgba('#646464', 0.4)};
  }

  ${applyStyleModifiers(modifierConfig)}

`;

const Button = (props) => {
  return (
    <StyledButton {...props}>
      {props.children}
    </StyledButton>
  )
};

// Button.Icon = Icon;
// Button.Text = Text;

export default withTheme(Button);
