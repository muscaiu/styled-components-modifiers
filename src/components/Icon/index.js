import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';

import modifierConfig from './modifierConfig';
import BasicIcon from './BasicIcon';

const Icon = styled(BasicIcon)`
  ${applyStyleModifiers(modifierConfig)}
`;

export default Icon;
