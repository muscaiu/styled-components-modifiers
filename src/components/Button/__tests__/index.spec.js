import React from 'react'
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import Button from 'components/Button'
import Icon from 'components/Icon'
import Text from 'components/Text'
import theme from 'theme';

function renderWithTheme(component) {
    return renderer.create(
        <ThemeProvider theme={theme}>
            {component}
        </ThemeProvider>
    );
};

it('renders correctly', () => {
    const tree = renderWithTheme(
        <Button>
            <Text>Login</Text>
            <Icon name="home" />
        </Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('adds the primary modifier', () => {
    const tree = renderWithTheme(
        <Button modifiers={['primary']}>
            <Text>Login</Text>
            <Icon name="home" />
        </Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('adds the secondary modifier', () => {
    const tree = renderWithTheme(
        <Button modifiers={['secondary']}>
            <Text>Login</Text>
            <Icon name="home" />
        </Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('adds the disabled modifier', () => {
    const tree = renderWithTheme(
        <Button modifiers={['disabled']}>
            <Text>Login</Text>
            <Icon name="home" />
        </Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});


