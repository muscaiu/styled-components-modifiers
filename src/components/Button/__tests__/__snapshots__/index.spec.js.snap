// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds the disabled modifier 1`] = `
.c0 {
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFF;
  cursor: pointer;
  font-size: 1rem;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  opacity: 50%;
  pointer-events: none;
}

.c0:hover,
.c0:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

.c2 {
  font-size: 1rem;
}

.c1 {
  font-size: 1rem;
  font-weight: 300;
  text-transform: uppercase;
}

<button
  className="c0"
>
  <span
    className="c1"
  >
    Login
  </span>
  <span
    aria-hidden={true}
    className="fa fa-home c2"
  />
</button>
`;

exports[`adds the primary modifier 1`] = `
.c0 {
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFF;
  cursor: pointer;
  font-size: 1rem;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  background-color: #7E5BEF;
}

.c0:hover,
.c0:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

.c2 {
  font-size: 1rem;
}

.c1 {
  font-size: 1rem;
  font-weight: 300;
  text-transform: uppercase;
}

<button
  className="c0"
>
  <span
    className="c1"
  >
    Login
  </span>
  <span
    aria-hidden={true}
    className="fa fa-home c2"
  />
</button>
`;

exports[`adds the secondary modifier 1`] = `
.c0 {
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFF;
  cursor: pointer;
  font-size: 1rem;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  background-color: #FFC82C;
}

.c0:hover,
.c0:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

.c2 {
  font-size: 1rem;
}

.c1 {
  font-size: 1rem;
  font-weight: 300;
  text-transform: uppercase;
}

<button
  className="c0"
>
  <span
    className="c1"
  >
    Login
  </span>
  <span
    aria-hidden={true}
    className="fa fa-home c2"
  />
</button>
`;

exports[`renders correctly 1`] = `
.c0 {
  background-color: #7D7D7D;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #FFF;
  cursor: pointer;
  font-size: 1rem;
  padding: 0.5rem 1.5rem;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c0:hover,
.c0:focus {
  background-color: #8a8a8a;
  box-shadow: 0 8px 8px 0 rgba(100,100,100,0.4);
}

.c2 {
  font-size: 1rem;
}

.c1 {
  font-size: 1rem;
  font-weight: 300;
  text-transform: uppercase;
}

<button
  className="c0"
>
  <span
    className="c1"
  >
    Login
  </span>
  <span
    aria-hidden={true}
    className="fa fa-home c2"
  />
</button>
`;
